OpenRA.Mods.Common/Effects/Bullet.cs

```diff
@@ -132,11 +130,32 @@ public void Tick(World world)
 				smokeTicks = info.TrailInterval;
 			}
 
+			var nextpos = WPos.LerpQuadratic(args.Source, target, angle, ticks, length);
+
+			if (!info.High)
+			{
+				var cell = world.Map.CellContaining(pos);
+				var nextcell = world.Map.CellContaining(nextpos);
+
+				var regularLength = (cell - nextcell).LengthSquared;
+
+				var collision = world.ActorMap.GetActorsOnRay(cell, nextcell).Where(x => x.HasTrait<IBlocksBullets>())
+					.FirstOrDefault(actor => !actor.IsDead && regularLength >= (cell - actor.Location).LengthSquared);
+
+				if (collision != null)
+				{
+					pos = collision.CenterPosition;
```

Comment:
I don't actually know what my collision point is, however. It is always going to be a cell large unless I add another algorithm/equation.

On 2015-03-22, at 11:03 AM, Paul Chote <notifications@github.com> wrote:

> In OpenRA.Mods.Common/Effects/Bullet.cs:
> 
> > @@ -132,11 +130,32 @@ public void Tick(World world)
> >  				smokeTicks = info.TrailInterval;
> >  			}
> >  
> > +			var nextpos = WPos.LerpQuadratic(args.Source, target, angle, ticks, length);
> > +
> > +			if (!info.High)
> > +			{
> > +				var cell = world.Map.CellContaining(pos);
> > +				var nextcell = world.Map.CellContaining(nextpos);
> > +
> > +				var regularLength = (cell - nextcell).LengthSquared;
> > +
> > +				var collision = world.ActorMap.GetActorsOnRay(cell, nextcell).Where(x => x.HasTrait<IBlocksBullets>())
> > +					.FirstOrDefault(actor => !actor.IsDead && regularLength >= (cell - actor.Location).LengthSquared);
> > +
> > +				if (collision != null)
> > +				{
> > +					pos = collision.CenterPosition;
> It would be better if this was the actual collision point, in case your collision actor is large than one cell. I guess that is what the regularLength stuff above is about?
> 
> \
> Reply to this email directly or view it on GitHub.
>